projects/gloo/pkg/translator/translator_test.go

```
@@ -1172,91 +1173,145 @@ var _ = Describe("Translator", func() {
-					HealthyThreshold:   DefaultThreshold,
-					UnhealthyThreshold: DefaultThreshold,
-					HealthChecker: &envoy_config_core_v3.HealthCheck_HttpHealthCheck_{
-						HttpHealthCheck: &envoy_config_core_v3.HealthCheck_HttpHealthCheck{
-							Host: "host",
-							Path: "path",
-							ServiceNameMatcher: &envoy_type_matcher_v3.StringMatcher{
-								MatchPattern: &envoy_type_matcher_v3.StringMatcher_Prefix{
-									Prefix: "svc",
-								},
+				var err error
+				upstream.HealthChecks, err = api_conversion.ToGlooHealthCheckList(expectedResult)
+				Expect(err).NotTo(HaveOccurred())
+				upstream.GetHealthChecks()[0].GetHttpHealthCheck().RequestHeadersToAdd = []*envoycore_sk.HeaderValueOption{
+					{
+						HeaderOption: &envoycore_sk.HeaderValueOption_HeaderSecretRef{
+							HeaderSecretRef: &core.ResourceRef{
+								Name:      "foo",
+								Namespace: "bad-tenant",
 							},
-							RequestHeadersToAdd:    []*envoy_config_core_v3.HeaderValueOption{},
-							RequestHeadersToRemove: []string{},
-							CodecClientType:        envoy_type_v3.CodecClientType_HTTP2,
-							ExpectedStatuses:       []*envoy_type_v3.Int64Range{},
+						},
+						Append: &wrappers.BoolValue{
+							Value: true,
 						},
 					},
-				},
-			}
+				}
 
-			var err error
-			upstream.HealthChecks, err = api_conversion.ToGlooHealthCheckList(expectedResult)
-			Expect(err).NotTo(HaveOccurred())
+				_, errs, _ := translator.Translate(params, proxy)
+				Expect(errs.Validate()).To(MatchError(ContainSubstring("list did not find secret bad-tenant.foo")))
 
-			expectedResult[0].GetHttpHealthCheck().RequestHeadersToAdd = []*envoy_config_core_v3.HeaderValueOption{
-				{
-					Header: &envoy_config_core_v3.HeaderValue{
-						Key:   "Authorization",
-						Value: "basic dXNlcjpwYXNzd29yZA==",
+			})
+			DescribeTable("can translate health check with secret header", func(enforceMatch, secretNamespace string) {
+				os.Setenv(api_conversion.MatchingNamespaceEnv, enforceMatch)
```

nit: we can check the swallowed error